From d578fc39f3ed5ac6d9836771af29bb946a604a7d Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 20 Feb 2021 08:55:51 -0500
Subject: [PATCH 3/3] REW: Implement per-process target SDK version override

  - based on @phhusson's version of Danny Baumann commit
  - adapted for LOS 18.1 GSI

Change-Id: Iadf6c14951d9376e088f5e0765d7a1993d961f1c
---
 linker/linker.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/linker/linker.cpp b/linker/linker.cpp
index 62d39fa53..fa0fa0a96 100644
--- a/linker/linker.cpp
+++ b/linker/linker.cpp
@@ -3597,7 +3597,9 @@ std::vector<android_namespace_t*> init_default_namespaces(const char* executable
   }
 
   uint32_t target_sdk = config->target_sdk_version();
-#ifdef SDK_VERSION_OVERRIDES
+
+  std::string sdkVersionOverrides = android::base::GetProperty("persist.sys.phh.sdk_override", "");
+  static const char *SDK_VERSION_OVERRIDES = sdkVersionOverrides.c_str();
   for (const auto& entry : android::base::Split(SDK_VERSION_OVERRIDES, " ")) {
     auto splitted = android::base::Split(entry, "=");
     if (splitted.size() == 2 && splitted[0] == executable_path) {
@@ -3606,7 +3608,7 @@ std::vector<android_namespace_t*> init_default_namespaces(const char* executable
     }
   }
   DEBUG("Target SDK for %s = %d", executable_path, target_sdk);
-#endif
+
   set_application_target_sdk_version(target_sdk);
 
   std::vector<android_namespace_t*> created_namespaces;
-- 
2.17.1

